### ISSUE TITLE
Training on custom dataset

### ISSUE THREAD

**Issue by himansh1314** [HobbitLong/SupContrast#69]
I was able to train the encoder model with projection head on a custom dataset, but when training the classification head with softmax activation, there is no option, it only supports training on two datasets. Can you check that, or am I missing something? I think you only need to add the option in argument parser, otherwise the dataloader is already built in other files.

**Comment by HobbitLong** [comment]
Originally it only supports CIFAR-10, and CIFAR-100. Some pull request added custom dataset. You could also easily added the custom dataset option to main_linear.py, and a pull request is welcome.

**Comment by RaphaelWag** [comment]
@HobbitLong 
In my Repo https://github.com/RaphaelWag/SupContrast I added the possibility to use custom data sets with all methods. Do you want to pull it?

**Comment by noreenanwar** [comment]
> I was able to train the encoder model with projection head on a custom dataset, but when training the classification head with softmax activation, there is no option, it only supports training on two datasets. Can you check that, or am I missing something? I think you only need to add the option in argument parser, otherwise the dataloader is already built in other files.

hi, can you please guide me on how did you train this model on a customized dataset?
